llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -16476,12 +16476,12 @@ SITargetLowering::getRegForInlineAsmConstraint(const TargetRegisterInfo *TRI_,
 
   const TargetRegisterClass *RC = nullptr;
```

Comment:
Everything in this file looks like unrelated style changes

Reply:
No, this is actually important to support the asm label constraint "!i". This should fall through to the default case (imho) without causing `VT.getSizeInBits()` to crash.

Reply:
Can you do this separately? You should be able to refer to labels in regular inline asm

Reply:
Apparently you can only refer to labels with callbr asm, otherwise you have to use ptr typed blockaddres values. 

But you can still commit this part separately, it's an existing issue independently of the label handling. e.g.

```
define void @func() {
  call void asm "", "v"({} poison)
  ret void
}
```

asserts here

Reply:
How should your example be lowered? The only examples of the empty struct I could find in `llvm/test/CodeGen/AMDGPU` are using it as a function parameter, where it's expected to be ignored (no register allocated for it).

Reply:
Probably should just be an error, this case does appear to crash all the other targets to in different places

Reply:
(done via https://github.com/llvm/llvm-project/pull/153425)